proto/net.proto

```diff
@@ -117,6 +119,14 @@ message NetworkFormat {
     NETWORK_SE = 2;
   }
   optional NetworkStructure network = 3;
+
+  // Policy output format of the NN.
+  enum PolicyFormat {
+    POLICY_UNKNOWN = 0;
+    POLICY_CLASSICAL = 1;
+    POLICY_CONVOLUTION = 2;
+  }
+  optional PolicyFormat policy = 4;
```

Comment:
I think any change in network structure should be a separate NetworkStructure enum value, rather than having a separate enum for that.
Otherwise code has to check all those enums, which is messy. And also older versions of a binary won't know at all that there are something to check.

So idea is:
- backends only look into `NetworkStructure`
- position encoder in search only looks into `InputFormat` (and as change in input format always change network structure, `NetworkStructure` should be incremented too). Currently not supported, so we only have one element there.
- The same with `OutputFormat`, only NN result decoder looks there.

Reply:
I put policy in separate enum because value and policy head structures can be different than the rest of the network. This policy head can be used with either SE or classical networks and same with WDL value head. Encoding all the possibilities in one NetworkStrucure enum would be very awkward.

Policy is always going to look the same 1858 element long vector to search code since those are all the legal moves in chess, so this enum is not something that search code should be aware of.

For older versions there is minimum version field that is set to 0.21 for networks with convolutional policy so that they don't try to decode it.

Reply:
Then I think it's still make sense to add one entry into `NetworkStructure` (something like `NETWORK_SE_WITH_POLICYFORMAT`), because otherwise old clients don't know that they have to look into PolicyFormat, and try to load something that they don't support.

Reply:
It's not enough to set the minimum engine version in protobuf with this policy head to 0.21?

Do I also add `NETWORK_CLASSICAL_WITH_POLICYFORMAT` for weights not using SE, but have AZ policy head? That's the network format that's closest to what AZ used and the code does support it right now.

Reply:
Minimal version is kind of the second line of defense.
It can happen (and does happen) that different backends support different set of features. E.g. traditionally tensorflow backend is behind. So just checking minimal version is not enough.